src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -232,6 +238,59 @@ private AspNetCore.HttpSys.Internal.SocketAddress LocalEndPoint
 
         internal WindowsPrincipal User { get; }
 
+        public SslProtocols Protocol { get; private set; }
+
+        public CipherAlgorithmType CipherAlgorithm { get; private set; }
+
+        public int CipherStrength { get; private set; }
+
+        public HashAlgorithmType HashAlgorithm { get; private set; }
+
+        public int HashStrength { get; private set; }
+
+        public ExchangeAlgorithmType KeyExchangeAlgorithm { get; private set; }
+
+        public int KeyExchangeStrength { get; private set; }
+
+        private void GetTlsHandshakeResults()
+        {
+            var handshake = _nativeRequestContext.GetTlsHandshake();
```

Comment:
Can the handshake be null here? Probably a good check to have anyways event if it can't be null.

Reply:
No, it's a struct